nestkernel/eprop_archiving_node.cpp

```diff
@@ -0,0 +1,300 @@
+  }
+}
+
+
+void
+nest::EpropArchivingNode::register_update( double t_last_update, double t_current_update )
+{
+  // register spike time if it is not in the list, otherwise increase access counter
+  std::vector< histentry_extended >::iterator it_reg =
+    std::lower_bound( t_last_update_per_synapse_.begin(), t_last_update_per_synapse_.end(), t_current_update - eps_ );
+
+  if ( it_reg == t_last_update_per_synapse_.end() || fabs( t_current_update - it_reg->t_ ) > eps_ )
+  {
+    t_last_update_per_synapse_.insert( it_reg, histentry_extended( t_current_update, 0.0, 1 ) );
+  }
+  else
+  {
+    ++it_reg->access_counter_;
+  }
+  // search for old entry, decrease access counter, and delete entry if the access counter equals zero
+  it_reg =
+    std::lower_bound( t_last_update_per_synapse_.begin(), t_last_update_per_synapse_.end(), t_last_update - eps_ );
+
+  if ( it_reg == t_last_update_per_synapse_.end() || fabs( t_last_update - it_reg->t_ ) > eps_ )
+  {
+    std::cout << "found no entry to register in scanned " + std::to_string( t_last_update ) + " ms interval \n";
+  }
+  else
+  {
+    it_reg->access_counter_--;
+
+    if ( it_reg->access_counter_ == 0 )
+    {
+      it_reg = t_last_update_per_synapse_.erase( it_reg ); // erase old entry
+    }
+  }
+}
+
+
+void
+nest::EpropArchivingNode::get_spike_history( double t1,
+  double t2,
+  std::deque< double >::iterator* start,
+  std::deque< double >::iterator* finish )
+{
+  // set pointer to spike history entries corresponding to times t1 and t2
+  *finish = spike_history_.end();
+
+  if ( spike_history_.empty() )
+  {
+    *start = *finish;
+    return;
+  }
+  else
+  {
+    std::deque< double >::iterator runner1 =
+      std::lower_bound( spike_history_.begin(), spike_history_.end(), t1 + eps_ );
+
+    *start = runner1;
+
+    std::deque< double >::iterator runner2 = std::lower_bound( runner1, spike_history_.end(), t2 + eps_ );
+
+    *finish = runner2;
+  }
+}
+
+void
+nest::EpropArchivingNode::tidy_eprop_history()
+{
+  double earliest_time_to_keep = ( t_last_update_per_synapse_.begin() )->t_;
+  if ( !eprop_history_.empty() )
+  {
+    std::deque< histentry_eprop >::iterator start;
+    std::deque< histentry_eprop >::iterator finish;
+
+    // find eprop history entries for times earlier than earliest last update time
+    nest::EpropArchivingNode::get_eprop_history( 0.0, earliest_time_to_keep, &start, &finish );
+
+    eprop_history_.erase( eprop_history_.begin(), finish ); // erase found entries since no longer used
+  }
+}
+
+void
+nest::EpropArchivingNode::tidy_spike_history()
+{
+  double earliest_time_to_keep = ( t_last_update_per_synapse_.begin() )->t_;
+  while ( ( !spike_history_.empty() ) && ( spike_history_.front() + 1.0e-6 < earliest_time_to_keep ) )
+  {
+    spike_history_.pop_front();
+  }
+}
+
+void
+nest::EpropArchivingNode::write_v_m_pseudo_deriv_to_eprop_history( long time_step, double v_m_pseudo_deriv )
+{
+  if ( n_incoming_ )
```

Comment:
Does this really make sense?

Reply:
Removed in commit https://github.com/nest/nest-simulator/pull/2867/commits/dfe95e07762785ae4715085e44a135532bd736e0